### PR TITLE
Fix(0618)(layerwise): dedup merged ONNX graph and keep key prefill passess enabled.

### PR DESCRIPTION
**Pull request by vbaddi**
Fixes two correctness/topology issues in the layerwise prefill export path:

  1. **Post-merge ONNX dedup (CSE).** The layerwise stitch merges per-window ONNX
     models, which duplicates equivalent pure-ONNX precompute chains (mask / RoPE
     prep) across windows. This adds a local common-subexpression-elimination pass
     (`_deduplicate_redundant_onnx_nodes`) that runs once on the final merged graph
     in `run_merge_pipeline`, canonicalizing duplicate nodes onto the first producer
     value.

  2. **Keep key prefill passes enabled during safe export.** `_jit_pass_cse`,
     `_jit_pass_constant_propagation`, and `_jit_pass_onnx_constant_fold` are added
     to `_SAFE_EXPORT_REQUIRED_PASSES` so they are *not* disabled during layerwise
     safe export. This reduces ONNX topology drift between layerwise prefill exports
     and regular (non-layerwise) prefill exports.

### CODE REVIEW

**Comment by ochougul** [comment]
tests done.
```
(qeff) ochougul@gb-283-blr-10:~/layerwise_new$ python examples/image_text_to_text/models/qwen3_5_moe/qwen3_5_moe_l
ayerwise_decode.py
/home/ochougul/envs/qeff/lib/python3.12/site-packages/onnxscript/converter.py:816: FutureWarning: 'onnxscript.values.Op.param_schemas' is deprecated in version 0.1 and will be removed in the future. Please use '.op_signature' instead.
  param_schemas = callee.param_schemas()
`torch_dtype` is deprecated! Use `dtype` instead!
`torch_dtype` is deprecated! Use `dtype` instead!
The fast path is not available because one of the required library is not installed. Falling back to torch implementation. To install follow https://github.com/fla-org/flash-linear-attention#installation and https://github.com/Dao-AILab/causal-conv1d
[Warning]: No transforms applied to model: Qwen3_5MoeForConditionalGeneration. It may be an unsupported model!
[Warning]: The subfunction feature is experimental. Please note that using compile consecutively with and without subfunction may produce inconsistent results.
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2979.26it/s]
Skipping module attribute 'freeze'
[Warning]: Converting a tensor to a Python number might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
[Warning]: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
[Warning]: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
[Warning]: Exporting aten::index operator of advanced indexing in opset 17 is achieved by combination of multiple ONNX operators, including Reshape, Transpose, Concat, and Gather. If indices include negative values, the exported graph will produce incorrect results.

Total export time: 21.50 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 1913.27it/s]
[Warning]: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!

Total export time: 19.36 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2611.61it/s]

Total export time: 18.32 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2162.87it/s]
[Warning]: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
[Warning]: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
[Warning]: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
[Warning]: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
[Warning]: ONNX Preprocess - Removing mutation from node aten::index_put_ on block input: '4'. This changes graph semantics. (Triggered internally at /pytorch/torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:348.)
[Warning]: ONNX Preprocess - Removing mutation from node aten::index_put_ on block input: '5'. This changes graph semantics. (Triggered internally at /pytorch/torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp:348.)

Total export time: 16.50 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2953.54it/s]

Total export time: 18.46 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2724.87it/s]

Total export time: 19.63 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 1930.72it/s]

Total export time: 19.27 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 1971.39it/s]

Total export time: 16.26 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2326.47it/s]

Total export time: 18.62 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2531.00it/s]

Total export time: 18.28 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2531.76it/s]

Total export time: 18.15 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2237.15it/s]

Total export time: 16.38 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2198.41it/s]

Total export time: 18.30 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 3095.87it/s]

Total export time: 18.15 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2522.14it/s]

Total export time: 18.54 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2611.30it/s]

Total export time: 16.22 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 1921.70it/s]

Total export time: 17.93 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2467.64it/s]

Total export time: 18.22 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2648.67it/s]

Total export time: 17.96 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2600.75it/s]

Total export time: 15.83 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2921.59it/s]

Total export time: 17.75 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2992.62it/s]

Total export time: 17.91 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2184.42it/s]

Total export time: 17.68 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2106.32it/s]

Total export time: 15.74 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2291.42it/s]

Total export time: 17.69 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 1880.88it/s]

Total export time: 17.68 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2082.22it/s]

Total export time: 17.81 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 1943.64it/s]

Total export time: 15.79 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2018.91it/s]

Total export time: 17.73 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2105.17it/s]

Total export time: 17.80 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2581.56it/s]

Total export time: 17.58 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2794.00it/s]

Total export time: 15.74 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2513.20it/s]

Total export time: 17.62 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 1962.20it/s]

Total export time: 17.53 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2681.63it/s]

Total export time: 17.89 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 2336.47it/s]

Total export time: 15.78 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2123.89it/s]

Total export time: 17.76 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 2649.76it/s]

Total export time: 17.75 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 354/354 [00:00<00:00, 3340.66it/s]

Total export time: 17.85 seconds
Loading weights: 100%|███████████████████████████████████████████████████████| 351/351 [00:00<00:00, 3433.82it/s]

Total export time: 26.48 seconds
Final QPC path: {'lang_decode_qpc_path': PosixPath('decode_qwen35_moe/Qwen3_5MoeForConditionalGeneration/Qwen3_5MoeDecoderWrapper-da834ad46dd74c44/qpc-a4ae168c6e4ec034/qpc')}
[[ 8160   579   264  7047  1817    25   271    16    13   220  2972  2014
  53983  2570  5396 64700   198   256   471  2570  2640    25   328 39113
    728   883  5941  1149   198   256   471  1061   369   264  4047  8306
   3296    11 10813  9859   364   264   638 19441 16250   466 22527   314
  16432    14 16417    13   271    17    13   220  2972 27382  1386  5141
  84457    14  1905 64700   198   256   471   353  1044  1167 16451    11
    264  3349  3992  1558  7633   539 52540  5559   579 48696 36814 11274
     13   198   256   471   353  1220  5707   303   264  2708    11 61446
     11   321 10631 11233     1]]
['Here\'s a thinking process:\n\n1.  **Analyze User Input:**\n   - User says: "Tell me about yourself."\n   - This is a common opening question, typically asking for a self-introduction or overview of capabilities/identity.\n\n2.  **Identify Key Constraints/Context:**\n   - I am Qwen, a large language model developed by Alibaba Group\'s Tongyi Lab.\n   - I should respond in a clear, concise, and helpful manner"']
Average Prefill time a.k.a TTFT is= 0.74 sec        
Decode is= 19.24 token/sec        
Total is= 16.96 token/sec        
Total (E2E) inference time is= 5.84 sec
```

**Review by ochougul — APPROVED** [review]
